### ISSUE TITLE
Opencart 4 and some Bugs

### ISSUE THREAD

**Issue by FelipoAntonoff** [opencart/opencart#11418]
Hello, today I tested Opencart 4 a little and found some bugs/details.
I tested the demo: https://demo.opencart.com/index.php?route=common/home&language=en-gb and I also installed Opencart 4 from scratch and had the same bugs.
Tested on Chrome version Version 103.0.5060.66 (Official version) 64 bits, used Desktop, but I also tested some on S10 Mobile and same bugs.

The videos are in Portuguese more to give you an idea and I summarize the details below.

Here's the test video with part of the bugs mentioned below:
[Vídeo 1](https://drive.google.com/file/d/1d7qFKLQUt9XYwv22YZsMzziYFKJKPuNX/view?usp=sharing)

Quick Test Login by the Administrator giving error in the demo:
[Vídeo 2](https://drive.google.com/file/d/1DhGzP6i3cD74-SHHYd23gVGvNX0I638s/view?usp=sharing)

Customer Registration Test in the Administrator, it even duplicated the customer:
[Vídeo 3](https://drive.google.com/file/d/1cqlggvKDfyHG4nxxKg1NM9EEpQrYSyMq/view?usp=sharing)

**The most serious/serious bugs follow in order:**
 * Add to Cart on the Home page or even on the Product page, the notice that was added appears, but it does not update in the Mini Cart and the Cart remains empty without products
  * Normal customer registration is not picking up and still cleans the data entered
  * Normal customer registration without displaying the telephone field and without the confirmation password (I don't know if it was removed)
  * In the search giving an error on the server, the error exposed:
  
```php
 Exception: Error: SQLSTATE[HY093]: Invalid parameter number: number of bound variables does not match number of tokens Error Code : HY093
SELECT COUNT(DISTINCT p.`product_id`) AS total FROM `product` p LEFT JOIN `product_description` pd ON (p.`product_id` = pd.`product_id`) LEFT JOIN `product_to_store` p2s ON (p.`product_id` = p2s.`product_id`) WHERE pd.`language_id` = '1' AND p.`status` = '1' AND p.`date_available` <= NOW() AND p2s.`store_id` = '0' AND ( pd.`name` LIKE '%:0%' OR pd.`tag` LIKE '%:1%' OR LCASE(p.`model`) = :2 OR LCASE(p.`sku`) = :3 OR LCASE(p.`upc`) = :4 OR LCASE(p.`ean`) = :5 OR LCASE(p.`jan`) = :6 OR LCASE(p.`isbn`) = :7 OR LCASE( p.`mpn`) = :8) in /home/.../public_html/oc4/system/library/db/pdo.php on line 55
```

  * Bug to login to the Customer account by the Administrator
  * I created a customer in the Administrator and created his registration twice with the same data, only the customer_id changed, including the email was the same, it shouldn't be possible to have 2 identical emails
  * Errors when trying to login:
```php
Warning: Undefined array key "address_id" in /home/../public_html/oc4/catalog/controller/account/login.php on line 212TypeError: Opencart\Catalog\Model\Account\Address::getAddress(): Argument #1 ( $address_id) must be of type int, null given in /home/.../public_html/oc4/catalog/model/account/address.php on line 28
```

I created the address and even then it kept error, then it came to log in with header error:
```php
Warning: Undefined array key "customer_token" in /home/.../public_html/oc4/catalog/controller/common/header.php on line 67Warning: Undefined array key "customer_token" in /home/.../public_html/oc4 /catalog/controller/common/header.php on line 68Warning: Undefined array key "customer_token" in /home/.../public_html/oc4/catalog/controller/common/header.php on line 69Warning: Undefined array key "customer_token" " in /home/.../public_html/oc4/catalog/controller/common/header.php on line 70
```

**Simpler bugs:**
  * Canon EOS 5D product in demo not being able to select Option, in my installation showing normal
  * Add to Cart button has black text when hovering, not visually pleasing
  * Mini cart (cart displayed in the header) has a black background in the text when hovering the mouse, but the grid background is already black, using the wrong color for the background of the letter on hover
  * Mini affection notice
  * Treat the customer's registration data better, in the Administrator I saw several with characters that should not be possible such as:74t7w9z$C 74t7w9z$C, the Customer's name should be just letters and space and the E-mail preferably only letters and numbers
  * Newsletter selection could be a similar checkbox as it is in Administrator (recommendation only)

**Recommendations:**
  * Take the demo off the air and use Opencart 3.0.3.8 which is more production ready
  * Undownload Opencart 4 from the site and leave only 3.0.3.8 and earlier, indicating 3.0.3.8 for production use
  * Remove Opencart version 4 from Marketplace in filter
  * Priority to fix these and other possible Opencart 4 bugs
  * The internal testing process should be improved, at least testing a purchase from scratch and by the Administrator for editing orders and changing the status of orders, which are essential uses of an E-commerce
  * Relaunch Opencart 4 once all relevant bugs are resolved and preferably after creating the new theme and essential features planned for it
  * Create a roadmap of releases and fixes release as soon as possible especially if they are bugs that affect a sale

If you need I'm available for testing and also helping with certain parts of the code.

**Comment by danielkerr** [comment]
u have just repeated every bug already listed

**Comment by FelipoAntonoff** [comment]
Hi @danielkerr , I hadn't seen it, but maybe I reported some not listed and I also reported some recommendations.

The testing process and the time for corrections needs to be improved, it's been more than 1 month since launch and even so it contains bugs that make a sale impossible.

I recommend you to carefully check the details I passed and the issues of other colleagues, I apologize if I repeated an issue, I tried to pass all the details I saw testing a few minutes of Opencart 4.

A colleague also commented on some Checkout details, these parts related to registration, sales, orders, cart and other points, we always have to pay special attention to the tests.
